Lib/test/test_msvcrt.py

```diff
@@ -64,35 +62,43 @@ def test_get_osfhandle(self):
 
 class TestConsoleIO(unittest.TestCase):
     def test_kbhit(self):
+        h = msvcrt.get_osfhandle(sys.stdin.fileno())
+        flush_console_input_buffer(h)
         self.assertEqual(msvcrt.kbhit(), 0)
 
     def test_getch(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getch(), b'c')
 
     def test_getwch(self):
-        stdin = open('CONIN$', 'r')
-        old_stdin = sys.stdin
-        try:
-            sys.stdin = stdin
-            write_input(stdin.buffer.raw, c_encoded)
-            self.assertEqual(msvcrt.getwch(), c)
-        finally:
-            sys.stdin = old_stdin
+        with open('CONIN$', 'rb', buffering=0) as stdin:
+            h = msvcrt.get_osfhandle(stdin.fileno())
+            flush_console_input_buffer(h)
+
+            old_stdin = sys.stdin
+            try:
+                sys.stdin = stdin
+                write_input(stdin, c_encoded)
+                self.assertEqual(msvcrt.getwch(), c)
+            finally:
+                sys.stdin = old_stdin
 
     def test_getche(self):
         msvcrt.ungetch(b'c')
         self.assertEqual(msvcrt.getche(), b'c')
 
     def test_getwche(self):
-        stdin = open('CONIN$', 'r')
-        old_stdin = sys.stdin
-        try:
-            sys.stdin = stdin
-            write_input(stdin.buffer.raw, c_encoded)
-            self.assertEqual(msvcrt.getwche(), c)
-        finally:
-            sys.stdin = old_stdin
+        with open('CONIN$', 'rb', buffering=0) as stdin:
+            h = msvcrt.get_osfhandle(stdin.fileno())
+            flush_console_input_buffer(h)
+
+            old_stdin = sys.stdin
+            try:
+                sys.stdin = stdin
```

Comment:
Updating `sys.stdin` shouldn't be necessary - the msvcrt.getwche() function uses the native stdin handle, not the Python object, and `write_input` takes the handle as an argument.

Reply:
Thanks for the review, updated.